enumerable_methods.rb

```diff
@@ -72,14 +78,15 @@ def my_all?
   end
 
   def my_any?
-    return to_enum unless block_given?
-
     arr = self
     if arr.is_a? Array
       i = 0
       while i < arr.length
-        return true if yield(arr[i])
-
+        if block_given?
+          return true if yield(arr[i])
+        else
+          return true if arr[i]
+        end
         i += 1
       end
       false
```

Comment:
**[REQUIRED]**
* When a class is passed as an argument returns true if at least one of the collection is a member of such class.
```
Failure/Error: expect(array.my_any?(Integer)).to be array.any?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns false if none of the collection matches the Regex.
```
Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```